contracts/teller-network/Escrow.sol

```diff
@@ -61,43 +70,113 @@ contract Escrow is Pausable, MessageSigned {
 
     enum ArbitrationResult {UNSOLVED, BUYER, SELLER}
 
+
     /**
      * @notice Create a new escrow
-     * @param _buyer The address that will perform the buy for the escrow
-     * @param _amount How much ether/tokens will be put in escrow
+     * @param _buyer Buyer address
+     * @param _seller Seller address
+     * @param _tradeAmount Amount buyer is willing to trade
+     * @param _tradeType Indicates if the amount is in crypto or fiat
      * @param _token Token address. Must be 0 for ETH
+     * @dev Requires contract to be unpaused.
+     *         The seller needs to be licensed.
+     */
+    function create(address payable _buyer, address payable _seller, address _token, uint _tradeAmount, uint8 _tradeType) public whenNotPaused {
+        require(msg.sender == _buyer || msg.sender == _seller, "Must participate in the trade");
+        require(license.isLicenseOwner(_seller), "Must be a valid seller to create escrow transactions");
+
+        uint escrowId = transactions.length++;
+
+        transactions[escrowId] = EscrowTransaction({
+            seller: _seller,
+            buyer: _buyer,
+            tokenAmount: 0,
+            token: _token,
+            expirationTime: 0,
+            rating: 0,
+            tradeAmount: _tradeAmount,
+            tradeType: TradeType(_tradeType),
+            status: EscrowStatus.CREATED
+        });
+
+        emit Created(_seller, _buyer, escrowId);
+    }
+
+    /**
+     * @notice Fund a new escrow
+     * @param _tokenAmount How much ether/tokens will be put in escrow
      * @param _expirationTime Unix timestamp before the transaction is considered expired
      * @dev Requires contract to be unpaused.
      *         The seller needs to be licensed.
      *         The expiration time must be at least 10min in the future
      *         For eth transfer, _amount must be equals to msg.value, for token transfer, requires an allowance and transfer valid for _amount
      */
-    function create(address payable _buyer, uint _amount, address _token, uint _expirationTime) public payable whenNotPaused {
+    function fund(uint _escrowId, uint _tokenAmount, uint _expirationTime) public payable whenNotPaused {
+        require(_escrowId < transactions.length, INVALID_ESCROW_ID);
         require(_expirationTime > (block.timestamp + 600), "Expiration time must be at least 10min in the future");
         require(license.isLicenseOwner(msg.sender), "Must be a valid seller to create escrow transactions");
```

Comment:
You don't verify that escrow seller and msg sender match?

Reply:
Now we do in line 121